helix-term/src/ui/editor.rs

```diff
@@ -330,7 +330,16 @@ impl EditorView {
         // document or not.  We only draw it if it's not an empty line.
         let draw_last = text.line_to_byte(last_line) < text.len_bytes();
 
-        for (i, line) in (view.first_line..=last_line).enumerate() {
+        // This is a little confuding: We can't use the `last_line` variable for
+        // our iteration below, because there's a mismatch between the
+        // inclusive/exclusiveness of the screen-based last line and the
+        // text-based last line.  So we compute what we actually need specially here.
+        let last_line_number = std::cmp::min(
+            view.first_line + view.area.height.saturating_sub(1) as usize,
```

Comment:
Why do we need `saturating_sub` when we already add it with something else? Can `first_line` be 0?

Reply:
`first_line` can definitely be zero, for example when the view is scrolled to the top of a text file.

Even so, making the subtraction saturating is *probably* not necessary, but it's good defensive practice in this case.  If some window manager lets the user resize their terminal window to zero height, for example.

Having said that, this is a bit of a hacky fix, and I'm not proud of it, ha ha.  Ideally we could just is `last_line` directly.  But it gets computed *slightly* differently because of how it's used elsewhere, and (I admit it) I was feeling too lazy to go and change all the other use-sites to match.

I'm planning to clean all of this up in my next big PR project, which will be implementing soft line wrapping (#136), etc.

Reply:
I decided to not be lazy.  I'm fixing `first_line` and its use-sites instead.  Ha ha.  Thanks for the poke on this.